### PR TITLE
Record Cypress test runs only on master branch

### PR DESCRIPTION
**Pull request by diogormendes**
I know we are not fully migrated to GHA still but already wanted to add the logic to run Currents only on Master.

### CODE REVIEW

File: .github/workflows/e2e-tests.yml [hunk 1]
```diff
@@ -112,9 +111,15 @@ jobs:
       run: |
         jar xf target/uberjar/metabase.jar version.properties
         mv version.properties resources/
-
     - name: Run Cypress tests on ${{ matrix.folder }}
-      run: yarn run test-cypress-no-build --folder ${{ matrix.folder }} --record --key ${{ secrets.CURRENTS_KEY }} --group ${{matrix.folder}} --ci-build-id "${{ github.run_id }}-${{ github.run_attempt }}"
+      run: |
```
Comment (ariya) [inline .github/workflows/e2e-tests.yml:115]: Simpler is to use `if`:

```
- name: Record Cypress tests on ${{ matrix.folder }} and send the report to Currents.dev
  if: github.ref == 'refs/heads/master'
```

**Review by flamber — COMMENTED** [review]
Is this still relevant?

**Comment by diogormendes** [comment]
> Is this still relevant?

Yes

File: .github/workflows/e2e-master.yml [hunk 2]
```diff
@@ -136,6 +136,13 @@ jobs:
           --ci-build-id "${{ github.run_id }}-${{ github.run_attempt }}"
       env:
         TERM: xterm
+    - name: Run Cypress tests on ${{ matrix.folder }}
+      if: ${{ github.ref != 'refs/heads/master' }}
+      run: |
+        yarn run test-cypress-no-build \
```
Comment (flamber) [inline .github/workflows/e2e-master.yml:142]:
```suggestion
        yarn run test-cypress-run \
```

File: .github/workflows/e2e-master.yml [hunk 3]
```diff
@@ -136,6 +136,13 @@ jobs:
           --ci-build-id "${{ github.run_id }}-${{ github.run_attempt }}"
       env:
         TERM: xterm
+    - name: Run Cypress tests on ${{ matrix.folder }}
```
Comment (flamber) [inline .github/workflows/e2e-master.yml:139]: Should this have a different name? I'm not sure how Github handles workflows with the same name - I'll defer to @ariya

**Review by flamber — COMMENTED** [review]
I don't know enough about GH workflows or Currents - but just want to make sure we aren't causing problems right now

File: .github/workflows/e2e-master.yml (hunk 3 above)
Comment (ariya) [reply .github/workflows/e2e-master.yml:139]: It's only for display purpose, shouldn't cause conflict or any terrible issue.

**Comment by nemanjaglumac** [comment]
@diogormendes can we close this PR since we already record on `e2e-master` workflow?

https://github.com/metabase/metabase/blob/master/.github/workflows/e2e-master.yml#L131:L136

**Comment by diogormendes** [comment]
Just added the comment on a Slack Thread about this. If we want to run on Release and will create a new workflow, OK to delete, if we think about having just one, we can rename this to something like `e2e-tests.yaml`

**Comment by nemanjaglumac** [comment]
>  if we think about having just one, we can rename this to something like `e2e-tests.yaml`

We can't because we already have a workflow with that name, but yeah - you're right. It would make sense to change the name to something else.

**Comment by nemanjaglumac** [comment]
@diogormendes now that #23166 is merged, you can rebase and use that renamed `e2e-main` workflow to apply these changes. Please keep two things in mind:

1. Add the glob for release branches
2. Please, please, please rename PR title to something like "Record Cypress test runs only on `master`" to make it more clear. Every time I see "Adding conditional to GHA E2E Tests" I have to remind myself what is it about. And in 6 months nobody will remember anymore.

Thanks

**Review by nemanjaglumac — APPROVED** [review]
Although I mentioned we should add the release branch glob here, it's even better if we don't (now when I think about it). Separation of concerns...

Let's merge this PR now and I'll open the next one immediately afterwards.

**Pull request merged**